src/HDInsight/HDInsightOnAks.Autorest/custom/New-AzHdInsightOnAksClusterConfigFile.ps1

```diff
@@ -0,0 +1,60 @@
+# ----------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# ----------------------------------------------------------------------------------
+
+
+function New-AzHdInsightOnAksClusterConfigFile {
+[OutputType([Microsoft.Azure.PowerShell.Cmdlets.HdInsightOnAks.Models.Api20230601Preview.IClusterConfigFile])]
+[CmdletBinding(DefaultParameterSetName='Create', PositionalBinding=$false)]
+param(
+    [Parameter(ParameterSetName='Create', Mandatory)]
+    [Alias('Name')]
+    [System.String]
+    # The name of the config file.
+    ${FileName},
+
+    [Parameter(ParameterSetName='Create', Mandatory)]
+    [System.Collections.HashTable]
+    ${Value}
+)
+
+begin {
+    try {
+        $configFileValue= New-Object Microsoft.Azure.PowerShell.Cmdlets.HdInsightOnAks.Models.Api20230601Preview.ClusterConfigFileValues
+
+        foreach($item in $Value.GetEnumerator()){
+            $configFileValue.Add($item.Name, $item.Value)
+        }
+
+        $configFile=New-Object Microsoft.Azure.PowerShell.Cmdlets.HdInsightOnAks.Models.Api20230601Preview.ClusterConfigFile -Property @{FileName=$FileName; Value=$configFileValue}
+        return $configFile
```

Comment:
Move the snippets to process {} block please

Reply:
Fixed